lib/code_owners.rb

```diff
@@ -35,11 +38,25 @@ def search_codeowners_file
     end
 
     # read the github file and spit out a slightly formatted list of patterns and their owners
+    # Empty/invalid/commented lines are still included in order to preserve line numbering
     def pattern_owners
       codeowner_path = search_codeowners_file
-      File.read(codeowner_path).split("\n").map do |line|
-        line.gsub(/#.*/, '').gsub(/^$/, " @").split(/\s+@/, 2)
-      end
+      i = 0
+      patterns = []
+      File.read(codeowner_path).split("\n").each { |line|
+        i += 1
+        path_owner = line.split(/\s+@/, 2)
+        if line.match(/^\s*#.*$/) || line.match(/^\s*$/)
+          patterns.push ['', ''] # Comment/empty line
+        elsif path_owner.length != 2 || path_owner[0].empty? && !path_owner[1].empty?
```

Comment:
I recommend wrapping your intended operator precedence with parens, helps with legibility, people not intimately familiar with Ruby's precedence, etc.  i.e. should it be `(A || B) && C` or `A || (B && C)`?